Add expand and allow custom root section options

josephrexme opened the pull request:
This addresses #1674 which allows user set and `expand` setting for each root section. It also sends all custom options from the user through to be usable

sapegin reviewed (changes requested):
Thanks for the pull request! Please update the docs and tests.

josephrexme commented:
Thanks for taking a look @sapegin . I have just added tests. The [syntax highlighting test here](https://github.com/styleguidist/react-styleguidist/pull/1689/files#diff-d56e2054ed9db0792599da8bcf6ce89b) that I modified was failing even though I hadn't touched it so I made it look like the value it was expecting. I can revert that if it's not needed. I don't know if I also had to include an updated snapshot

sapegin commented:
> The syntax highlighting test here that I modified was failing even though I hadn't touched it so I made it look like the value it was expecting. I can revert that if it's not needed. I don't know if I also had to include an updated snapshot

This is because Prism was updated at some point, and the snapshot is already fixed in master.

sapegin commented on docs/Configuration.md:
```diff
@@ -136,6 +136,12 @@ Defines the initial state of the example code tab:
 - `hide`: hide the tab and it can´t be toggled in the UI.
 - `expand`: expand the tab by default.
 
+## `expand`
```
I'm confused: this isn't a global configuration option, right? If so, we should document it here:

https://github.com/styleguidist/react-styleguidist/blob/master/docs/Components.md#sections

josephrexme replied:
Thank you!

sapegin commented on src/loaders/utils/getSections.ts:
```diff
@@ -88,6 +88,7 @@ export function processSection(
 
 	return {
+		...section,
 		name: section.name,
```
We don't need this line anymore, and also `description` and `external` below.

josephrexme commented:
> > The syntax highlighting test here that I modified was failing even though I hadn't touched it so I made it look like the value it was expecting. I can revert that if it's not needed. I don't know if I also had to include an updated snapshot
> 
> This is because Prism was updated at some point, and the snapshot is already fixed in master.

To be clear, should I undo my change with the **highlightCode.spec.ts** file?

sapegin commented:
Try to merge master into your branch.

josephrexme commented:
Thanks for the reviews and accepting my PR @sapegin
